tachyon/base/auto_reset.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2011 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef TACHYON_BASE_AUTO_RESET_H_
+#define TACHYON_BASE_AUTO_RESET_H_
+
+#include <utility>
+
+#include "tachyon/base/logging.h"
+// #include "base/memory/raw_ptr_exclusion.h"
+
+// base::AutoReset<> is useful for setting a variable to a new value only within
+// a particular scope. An base::AutoReset<> object resets a variable to its
+// original value upon destruction, making it an alternative to writing
+// "var = false;" or "var = old_val;" at all of a block's exit points.
+//
+// This should be obvious, but note that an base::AutoReset<> instance should
+// have a shorter lifetime than its scoped_variable, to prevent invalid memory
+// writes when the base::AutoReset<> object is destroyed.
+
+namespace tachyon::base {
+
+template <typename T>
+class [[maybe_unused, nodiscard]] AutoReset {
```

Comment:
Hmm, unittest is not related actually since this affects where this header is included but it isn't used. In our case, it's okay to be without this, but to minimize the diffs from chromium, I want it to be maintained!